@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -479,16 +479,28 @@ export class TextElementsRenderer {
      * Updates the FontCatalogs used by this {@link TextElementsRenderer}.
      *
      * @param fontCatalogs - The new list of {@link FontCatalogConfig}s
-     * @param overrideDefault - Optional, if `true` removes default catalog @default `false`.
      */
-    async updateFontCatalogs(fontCatalogs?: FontCatalogConfig[], overrideDefault: boolean = false) {
-        if (this.m_defaultFontCatalogConfig && !overrideDefault) {
-            // Never remove the default Canvas if set per configuration
-            if (!fontCatalogs) {
-                fontCatalogs = [];
+    async updateFontCatalogs(fontCatalogs?: FontCatalogConfig[]) {
+        if (this.m_defaultFontCatalogConfig) {
+            if (
+                !fontCatalogs ||
+                fontCatalogs.findIndex(config => {
+                    return config.name === DEFAULT_FONT_CATALOG_NAME;
+                }) === -1
+            ) {
+                // not other default catalog available, keep the old one
+                if (!fontCatalogs) {
+                    fontCatalogs = [];
+                }
+                // Never remove the default Canvas if set per configuration
+                fontCatalogs?.unshift(this.m_defaultFontCatalogConfig);
```

Comment:
fontCatalogs!.unshift(this.m_defaultFontCatalogConfig);

or even:

fontCatalogs.unshift(this.m_defaultFontCatalogConfig);

if the compiler is ok with that.

Reply:
done